autoPyTorch/pipeline/components/training/trainer/base_trainer.py

```diff
@@ -236,11 +273,25 @@ def on_epoch_end(self, X: Dict[str, Any], epoch: int) -> bool:
         """
         return False
 
+    def _scheduler_step(
+        self,
+        step_interval: StepIntervalUnit,
+        loss: Optional[torch.Tensor] = None
+    ) -> None:
+        if self.step_unit != step_interval:
+            return
+
+        if self.scheduler:
+            if 'ReduceLROnPlateau' in self.scheduler.__class__.__name__:
```

Comment:
I think it will be better not to hardcode things in this, and let every scheduler decide what type of step it should take?

Reply:
Yeah, I know what you mean, but it was the original code and I cannot guarantee the original behavior by changing here, so I would like to keep this for now.
Of course, if you have any good idea to solve this, I may adapt here immediately.